test/textSanitiserTest.js

```diff
@@ -0,0 +1,68 @@
+import { expect } from 'chai';
+import sanitise from '../src/textSanitiser';
+
+describe.only('error sanitiser', () => {
+  describe('the error message does not contain anything that looks sensitive', () => {
+    const text = 'foo is blah pass word cvx cord number';
+
+    it('does not change the error message', () => expect(sanitise(text)).to.equal(text));
+  });
+
+  describe('the error message contains something that looks like a card number', () => {
+    describe('the card number is blank', () => {
+      const text = '"foo":{"bar":132,"cardNumber":""';
+
+      it('does not insert any stars', () =>
+        expect(sanitise(text)).to.equal('"foo":{"bar":132,"cardNumber":""'));
+    });
+
+    describe('the card number is not blank', () => {
+      const text = '"foo":{"bar":132,"cardNumber":"12345678"';
+
+      it('replaces all but the last 4 chars with stars', () =>
+        expect(sanitise(text)).to.equal('"foo":{"bar":132,"cardNumber":"****5678"'));
+    });
+  });
+
```

Comment:
the way I test these things, to avoid repeating in and out params:

```es6
describe('the error message contains something that looks like a card number', () => {
  let params;
  let sanitize = () => sanitise(JSON.stringify(params));

  beforeEach('make some params', () => (
    params = {
      foo: {
        bar: null,
        cardNumber: null;
        cvv: null,
      },
    }
  ));
      
  context('when the card number is blank', () => {
    it('does not insert any stars', () => {
      params.foo.cardNumber = "";
      expect(sniatize()).to.not.include('*');
    });
  });

  context('when the card number is not blank', () => {
    it('replaces all but the last 4 chars with stars', () => {
      params.foo.cardNumber = 12345678;
      expect(sanitize()).to.include('"****5678"');
    });
  });
});
```